unified-prototype/unified-plugin/plugin-android/src/main/java/org/gradle/api/experimental/android/AndroidSoftwareBuildTypes.java

```diff
@@ -0,0 +1,13 @@
+package org.gradle.api.experimental.android;
+
+import org.gradle.api.tasks.Nested;
+import org.gradle.declarative.dsl.model.annotations.Restricted;
+
+@Restricted
+public interface AndroidSoftwareBuildTypes {
+    @Nested
```

Comment:
Can we try to remove `@Nested` from these `AndroidSoftware*` interfaces? Since we don't want Gradle to instantiate instances of these super-interfaces. The Application and Library sub-interfaces should declare `@Nested` with their more narrow type